File: app/src/main/java/vn/loitp/app/activity/ads/admobinterstitial/AdmobInterstitialActivity.java
package vn.loitp.app.activity.ads.admobinterstitial;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import com.google.android.gms.ads.InterstitialAd;

import vn.loitp.core.base.BaseActivity;
import loitp.basemaster.R;
import vn.loitp.core.utilities.LStoreUtil;
import vn.loitp.core.utilities.LUIUtil;

public class AdmobInterstitialActivity extends BaseActivity {
    private InterstitialAd interstitialAd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        isShowAdWhenExist = false;//remove show duplicate ads in BaseActivity

        interstitialAd = LUIUtil.createAdFull(activity);

        TextView tv = (TextView) findViewById(R.id.tv);
        String s = LStoreUtil.readTxtFromRawFolder(activity, R.raw.ad_full);
        tv.setText(s);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_admob_interstitial;
    }

    @Override
    public void onBackPressed() {
        LUIUtil.displayInterstitial(interstitialAd);
        super.onBackPressed();
    }
}

File: app/src/main/java/vn/loitp/app/activity/ads/admobinterstitial/AdmobInterstitialActivity.kt
package vn.loitp.app.activity.ads.admobinterstitial

import android.os.Bundle
import com.annotation.IsFullScreen
import com.annotation.LayoutId
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LStoreUtil
import com.core.utilities.LUIUtil
import com.google.android.gms.ads.InterstitialAd
import kotlinx.android.synthetic.main.activity_admob_interstitial.*
import vn.loitp.app.R

@LayoutId(R.layout.activity_admob_interstitial)
@LogTag("AdmobInterstitialActivity")
@IsFullScreen(false)
class AdmobInterstitialActivity : BaseFontActivity() {
    private var interstitialAd: InterstitialAd? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        interstitialAd = LUIUtil.createAdFull(this)
        val s = LStoreUtil.readTxtFromRawFolder(nameOfRawFile = R.raw.ad_full)
        textView.text = s
    }

    override fun onBackPressed() {
        LUIUtil.displayInterstitial(interstitialAd)
        super.onBackPressed()
    }
}
